code/obj/item/implant.dm

```diff
@@ -2143,14 +2143,41 @@ TYPEINFO(/obj/item/gun/implanter)
 
 	throw_impact(atom/M, datum/thrown_thing/thr)
 		..()
-		if (ishuman(M) && prob(5))
+		if (istype(M, /obj/item/reagent_containers/balloon))
+			var/obj/item/reagent_containers/balloon/balloon = M
+			balloon.smash()
+
+		else if (ishuman(M) && prob(5))
 			var/mob/living/carbon/human/H = M
 			H.implant.Add(src)
 			src.visible_message(SPAN_ALERT("[src] gets embedded in [M]!"))
 			playsound(src.loc, 'sound/impact_sounds/Flesh_Cut_1.ogg', 100, 1)
 			random_brute_damage(M, 1)
 			src.implanted(M)
 
+	hit_check(datum/thrown_thing/thr)
+		if(src.throwing)
+			for (var/atom/A as anything in get_turf(src))
+				if (!src.throwing)
+					break
+				if(A == src) continue
+				if(A.GetComponent(/datum/component/glued)) continue
+				if(isliving(A))
+					var/mob/living/L = A
+					if (!L.throws_can_hit_me) continue
+					if (L.lying) continue
+					src.throw_impact(A, thr)
+					. = TRUE
+				if(isobj(A))
+					if(!A.Cross(src) || istype(A, /obj/item/reagent_containers/balloon)) // darts can hit balloons too
+						src.throw_impact(A, thr)
+						. = TRUE
+				if(ishuman(src) && istype(A, /obj/item/plant/tumbling_creeper))
+					var/obj/item/plant/tumbling_creeper/M = A
+					if(M.armed)
+						src.throw_impact(M, thr)
+						. = TRUE
```

Comment:
Don't just copy paste the entire `/atom/movable/hit_check` here. A better way would be to implement `Cross` on balloons to deny thrown darts, that way the parent would catch it anyway.